{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f260157",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('file.txt', 'wb') as file:\n",
    "    file.write(b\"Ali, 12-3-20, S1, S2,S3\\n\n",
    "    Fahad 14-04-22, S1, S2, S7,S3\\n\n",
    "    Ali 20-4-22 S1, S2\\n\n",
    "    Sara 20-4-22 S1, S7, S9\\n\n",
    "    Amina 20-4-22 S1\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "497e3076",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Username</th>\n",
       "      <th>Date</th>\n",
       "      <th>list of song IDs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ali</td>\n",
       "      <td>12-3-20</td>\n",
       "      <td>S1, S2, S3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Fahad</td>\n",
       "      <td>14-04-22</td>\n",
       "      <td>S1, S2, S7, S3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ali</td>\n",
       "      <td>20-4-22</td>\n",
       "      <td>S1, S2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sara</td>\n",
       "      <td>20-4-22</td>\n",
       "      <td>S1, S7, S9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Amina</td>\n",
       "      <td>20-4-22</td>\n",
       "      <td>S1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Username      Date list of song IDs\n",
       "0      Ali   12-3-20       S1, S2, S3\n",
       "1    Fahad  14-04-22   S1, S2, S7, S3\n",
       "2      Ali   20-4-22           S1, S2\n",
       "3     Sara   20-4-22       S1, S7, S9\n",
       "4    Amina   20-4-22               S1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# import pandas library\n",
    "import pandas as pd\n",
    "\n",
    "# create a list of lists from the given data\n",
    "data = [\n",
    "    [\"Ali\", \"12-3-20\", \"S1, S2, S3\"],\n",
    "    [\"Fahad\", \"14-04-22\", \"S1, S2, S7, S3\"],\n",
    "    [\"Ali\", \"20-4-22\", \"S1, S2\"],\n",
    "    [\"Sara\", \"20-4-22\", \"S1, S7, S9\"],\n",
    "    [\"Amina\", \"20-4-22\", \"S1\"]\n",
    "]\n",
    "\n",
    "# create a list of column names\n",
    "columns = [\"Username\", \"Date\", \"list of song IDs\"]\n",
    "\n",
    "# create a data frame from the list of lists and the column names\n",
    "df = pd.DataFrame(data, columns=columns)\n",
    "\n",
    "# print the data frame\n",
    "display(df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ff31f7b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 634, in run_job\n",
      "    with self.make_runner() as runner:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 713, in make_runner\n",
      "    return self._runner_class()(**self._runner_kwargs())\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\inline.py\", line 67, in __init__\n",
      "    super(InlineMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 78, in __init__\n",
      "    super(SimMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 210, in __init__\n",
      "    load_opts_from_mrjob_confs(self.alias, conf_paths) +\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 325, in load_opts_from_mrjob_confs\n",
      "    results = load_opts_from_mrjob_conf(runner_alias)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 259, in load_opts_from_mrjob_conf\n",
      "    conf_path = _expanded_mrjob_conf_path(conf_path)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 85, in _expanded_mrjob_conf_path\n",
      "    return find_mrjob_conf()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 68, in find_mrjob_conf\n",
      "    log.info('No configs found; falling back on auto-configuration')\n",
      "Message: 'No configs found; falling back on auto-configuration'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 634, in run_job\n",
      "    with self.make_runner() as runner:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 713, in make_runner\n",
      "    return self._runner_class()(**self._runner_kwargs())\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\inline.py\", line 67, in __init__\n",
      "    super(InlineMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 78, in __init__\n",
      "    super(SimMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 210, in __init__\n",
      "    load_opts_from_mrjob_confs(self.alias, conf_paths) +\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 325, in load_opts_from_mrjob_confs\n",
      "    results = load_opts_from_mrjob_conf(runner_alias)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 259, in load_opts_from_mrjob_conf\n",
      "    conf_path = _expanded_mrjob_conf_path(conf_path)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 85, in _expanded_mrjob_conf_path\n",
      "    return find_mrjob_conf()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 68, in find_mrjob_conf\n",
      "    log.info('No configs found; falling back on auto-configuration')\n",
      "Message: 'No configs found; falling back on auto-configuration'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 634, in run_job\n",
      "    with self.make_runner() as runner:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 713, in make_runner\n",
      "    return self._runner_class()(**self._runner_kwargs())\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\inline.py\", line 67, in __init__\n",
      "    super(InlineMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 78, in __init__\n",
      "    super(SimMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 210, in __init__\n",
      "    load_opts_from_mrjob_confs(self.alias, conf_paths) +\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 325, in load_opts_from_mrjob_confs\n",
      "    results = load_opts_from_mrjob_conf(runner_alias)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 259, in load_opts_from_mrjob_conf\n",
      "    conf_path = _expanded_mrjob_conf_path(conf_path)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 85, in _expanded_mrjob_conf_path\n",
      "    return find_mrjob_conf()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 68, in find_mrjob_conf\n",
      "    log.info('No configs found; falling back on auto-configuration')\n",
      "Message: 'No configs found; falling back on auto-configuration'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 634, in run_job\n",
      "    with self.make_runner() as runner:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 713, in make_runner\n",
      "    return self._runner_class()(**self._runner_kwargs())\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\inline.py\", line 67, in __init__\n",
      "    super(InlineMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 78, in __init__\n",
      "    super(SimMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 210, in __init__\n",
      "    load_opts_from_mrjob_confs(self.alias, conf_paths) +\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 339, in load_opts_from_mrjob_confs\n",
      "    log.warning('No configs specified for %s runner' % runner_alias)\n",
      "Message: 'No configs specified for inline runner'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 634, in run_job\n",
      "    with self.make_runner() as runner:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 713, in make_runner\n",
      "    return self._runner_class()(**self._runner_kwargs())\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\inline.py\", line 67, in __init__\n",
      "    super(InlineMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 78, in __init__\n",
      "    super(SimMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 210, in __init__\n",
      "    load_opts_from_mrjob_confs(self.alias, conf_paths) +\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 339, in load_opts_from_mrjob_confs\n",
      "    log.warning('No configs specified for %s runner' % runner_alias)\n",
      "Message: 'No configs specified for inline runner'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 634, in run_job\n",
      "    with self.make_runner() as runner:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 713, in make_runner\n",
      "    return self._runner_class()(**self._runner_kwargs())\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\inline.py\", line 67, in __init__\n",
      "    super(InlineMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 78, in __init__\n",
      "    super(SimMRJobRunner, self).__init__(**kwargs)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 210, in __init__\n",
      "    load_opts_from_mrjob_confs(self.alias, conf_paths) +\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\conf.py\", line 339, in load_opts_from_mrjob_confs\n",
      "    log.warning('No configs specified for %s runner' % runner_alias)\n",
      "Message: 'No configs specified for inline runner'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 503, in run\n",
      "    self._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 146, in _run\n",
      "    self._output_dir = join(self._get_local_tmp_dir(), 'output')\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 732, in _get_local_tmp_dir\n",
      "    log.info('Creating temp directory %s' % path)\n",
      "Message: 'Creating temp directory C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 503, in run\n",
      "    self._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 146, in _run\n",
      "    self._output_dir = join(self._get_local_tmp_dir(), 'output')\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 732, in _get_local_tmp_dir\n",
      "    log.info('Creating temp directory %s' % path)\n",
      "Message: 'Creating temp directory C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 503, in run\n",
      "    self._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 146, in _run\n",
      "    self._output_dir = join(self._get_local_tmp_dir(), 'output')\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 732, in _get_local_tmp_dir\n",
      "    log.info('Creating temp directory %s' % path)\n",
      "Message: 'Creating temp directory C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 503, in run\n",
      "    self._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 156, in _run\n",
      "    log.info('Running step %d of %d...' % (\n",
      "Message: 'Running step 1 of 1...'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 503, in run\n",
      "    self._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 156, in _run\n",
      "    log.info('Running step %d of %d...' % (\n",
      "Message: 'Running step 1 of 1...'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 503, in run\n",
      "    self._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\sim.py\", line 156, in _run\n",
      "    log.info('Running step %d of %d...' % (\n",
      "Message: 'Running step 1 of 1...'\n",
      "Arguments: ()\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 510, in run\n",
      "    log.info('job output is in %s' % self._output_dir)\n",
      "Message: 'job output is in C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252\\\\output'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 510, in run\n",
      "    log.info('job output is in %s' % self._output_dir)\n",
      "Message: 'job output is in C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252\\\\output'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 636, in run_job\n",
      "    runner.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 510, in run\n",
      "    log.info('job output is in %s' % self._output_dir)\n",
      "Message: 'job output is in C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252\\\\output'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 644, in run_job\n",
      "    for chunk in runner.cat_output():\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 534, in cat_output\n",
      "    log.info('Streaming final output from %s...' % output_dir)\n",
      "Message: 'Streaming final output from C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252\\\\output...'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 644, in run_job\n",
      "    for chunk in runner.cat_output():\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 534, in cat_output\n",
      "    log.info('Streaming final output from %s...' % output_dir)\n",
      "Message: 'Streaming final output from C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252\\\\output...'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 644, in run_job\n",
      "    for chunk in runner.cat_output():\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 534, in cat_output\n",
      "    log.info('Streaming final output from %s...' % output_dir)\n",
      "Message: 'Streaming final output from C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252\\\\output...'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 645, in run_job\n",
      "    self.stdout.write(chunk)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 646, in run_job\n",
      "    self.stdout.flush()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 677, in __exit__\n",
      "    self.cleanup()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 652, in cleanup\n",
      "    self._cleanup_local_tmp()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 596, in _cleanup_local_tmp\n",
      "    log.info('Removing temp directory %s...' % self._local_tmp_dir)\n",
      "Message: 'Removing temp directory C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252...'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 645, in run_job\n",
      "    self.stdout.write(chunk)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 646, in run_job\n",
      "    self.stdout.flush()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 677, in __exit__\n",
      "    self.cleanup()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 652, in cleanup\n",
      "    self._cleanup_local_tmp()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 596, in _cleanup_local_tmp\n",
      "    log.info('Removing temp directory %s...' % self._local_tmp_dir)\n",
      "Message: 'Removing temp directory C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252...'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 645, in run_job\n",
      "    self.stdout.write(chunk)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\logging\\__init__.py\", line 1086, in emit\n",
      "    stream.write(msg + self.terminator)\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\codecs.py\", line 378, in write\n",
      "    self.stream.write(data)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py\", line 546, in write\n",
      "    raise TypeError(msg)\n",
      "TypeError: write() argument must be str, not <class 'bytes'>\n",
      "Call stack:\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 197, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel_launcher.py\", line 17, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\traitlets\\config\\application.py\", line 1041, in launch_instance\n",
      "    app.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelapp.py\", line 724, in start\n",
      "    self.io_loop.start()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\tornado\\platform\\asyncio.py\", line 215, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n",
      "    self._run_once()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n",
      "    handle._run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\asyncio\\events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 512, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 501, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 408, in dispatch_shell\n",
      "    await result\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\kernelbase.py\", line 731, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\ipkernel.py\", line 417, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\zmqshell.py\", line 540, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2945, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3000, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3203, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3382, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 3442, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\AB\\AppData\\Local\\Temp\\ipykernel_10152\\950126040.py\", line 1, in <module>\n",
      "    get_ipython().run_line_magic('run', 'user_songs.py song_list.txt')\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2369, in run_line_magic\n",
      "    result = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 829, in run\n",
      "    run()\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\magics\\execution.py\", line 814, in run\n",
      "    runner(filename, prog_ns, prog_ns,\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\core\\interactiveshell.py\", line 2802, in safe_execfile\n",
      "    py3compat.execfile(\n",
      "  File \"C:\\Users\\AB\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython\\utils\\py3compat.py\", line 55, in execfile\n",
      "    exec(compiler(f.read(), fname, \"exec\"), glob, loc)\n",
      "  File \"D:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py\", line 20, in <module>\n",
      "    MRJob_Songs.run()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 616, in run\n",
      "    cls().execute()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 687, in execute\n",
      "    self.run_job()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\job.py\", line 646, in run_job\n",
      "    self.stdout.flush()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 677, in __exit__\n",
      "    self.cleanup()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 652, in cleanup\n",
      "    self._cleanup_local_tmp()\n",
      "  File \"C:\\Users\\AB\\anaconda3\\lib\\site-packages\\mrjob\\runner.py\", line 596, in _cleanup_local_tmp\n",
      "    log.info('Removing temp directory %s...' % self._local_tmp_dir)\n",
      "Message: 'Removing temp directory C:\\\\Users\\\\AB\\\\AppData\\\\Local\\\\Temp\\\\user_songs.AB.20240220.073847.991252...'\n",
      "Arguments: ()\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "write() argument must be str, not <class 'bytes'>",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "File \u001b[1;32mD:\\my-folder\\Uni\\bs data science\\4\\Big Data\\Lab\\Manuals\\user_songs.py:20\u001b[0m\n\u001b[0;32m     18\u001b[0m \u001b[38;5;66;03m# run the job\u001b[39;00m\n\u001b[0;32m     19\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;18m__name__\u001b[39m \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m__main__\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m---> 20\u001b[0m     \u001b[43mMRJob_Songs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\mrjob\\job.py:616\u001b[0m, in \u001b[0;36mMRJob.run\u001b[1;34m(cls)\u001b[0m\n\u001b[0;32m    603\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Entry point for running job from the command-line.\u001b[39;00m\n\u001b[0;32m    604\u001b[0m \n\u001b[0;32m    605\u001b[0m \u001b[38;5;124;03mThis is also the entry point when a mapper or reducer is run\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    613\u001b[0m \u001b[38;5;124;03m* Run the entire job. See :py:meth:`run_job`\u001b[39;00m\n\u001b[0;32m    614\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m    615\u001b[0m \u001b[38;5;66;03m# load options from the command line\u001b[39;00m\n\u001b[1;32m--> 616\u001b[0m \u001b[38;5;28;43mcls\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexecute\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\mrjob\\job.py:687\u001b[0m, in \u001b[0;36mMRJob.execute\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    684\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrun_spark(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions\u001b[38;5;241m.\u001b[39mstep_num)\n\u001b[0;32m    686\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 687\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun_job\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\mrjob\\job.py:645\u001b[0m, in \u001b[0;36mMRJob.run_job\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    643\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_should_cat_output():\n\u001b[0;32m    644\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m chunk \u001b[38;5;129;01min\u001b[39;00m runner\u001b[38;5;241m.\u001b[39mcat_output():\n\u001b[1;32m--> 645\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mstdout\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrite\u001b[49m\u001b[43m(\u001b[49m\u001b[43mchunk\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    646\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstdout\u001b[38;5;241m.\u001b[39mflush()\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\ipykernel\\iostream.py:546\u001b[0m, in \u001b[0;36mOutStream.write\u001b[1;34m(self, string)\u001b[0m\n\u001b[0;32m    544\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(string, \u001b[38;5;28mstr\u001b[39m):\n\u001b[0;32m    545\u001b[0m     msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mwrite() argument must be str, not \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mtype\u001b[39m(string)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m--> 546\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(msg)\n\u001b[0;32m    548\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mecho \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m    549\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "\u001b[1;31mTypeError\u001b[0m: write() argument must be str, not <class 'bytes'>"
     ]
    }
   ],
   "source": [
    "%run user_songs.py song_list.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5ef7101",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
